### PR TITLE
feat: support locale-based collations when sorting (MX-7704)

### PR DESCRIPTION
**Pull request by jsalvata**
This allows sorting 'a' < 'B' < 'c' < 'D' instead of 'B' < 'D' < 'a' < 'c', or '2' < '9' < '10' instead of '10' < '2' < '9'  by configuring a collation to use for sorting (see https://docs.mongodb.com/manual/reference/collation/ ).

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,7 +1,9 @@
 ## Changelog
 
+* 7.2.0 Add support for `COLLATION` configuration parameter.
```
Comment (sbalaji1996) [inline CHANGELOG.md]: Do we need to update the changelog manually (i.e. will semantic release update this for us automatically)?

File: package.json [hunk 2]
```diff
@@ -55,7 +55,7 @@
     "mockgoose": "7.3.5",
     "mongodb": "^2.2.11",
     "mongodb-memory-server": "^1.6.1",
-    "mongoist": "1.4.3",
+    "mongoist": "2.3.0",
```
Comment (sbalaji1996) [inline package.json]: Are we concerned with what necessitated the breaking change?

File: src/find.js [hunk 3]
```diff
@@ -1,6 +1,7 @@
 const _ = require('underscore');
 const sanitizeParams = require('./utils/sanitizeParams');
 const { prepareResponse, generateSort, generateCursorQuery } = require('./utils/query');
+var config = require('./config');
```
Comment (sbalaji1996) [inline src/find.js]: I think we generally prefer `const` and `let` over `var`

```suggestion
const config = require('./config');
```

File: CHANGELOG.md (hunk 1 above)
Comment (jsalvata) [reply CHANGELOG.md]: I can't tell. I'll ask Eli.

File: package.json (hunk 2 above)
Comment (jsalvata) [reply package.json]: Good point. It's the requirement to specify a port in the connection string: https://github.com/mongoist/mongoist/commit/90dd80bd1cfe812be0746ce99f6ec15a886f15d5

Note however that this is in `devDependencies`. The actual version of `mongoist` being used is specified in the dependent package -- `compose` in our case. We are moving from 2.0.0 to 2.3.0 there: https://github.com/mixmaxhq/compose/pull/8243/files#diff-b9cfc7f2cdf78a7f4b91a753d10865a2

Comment (sbalaji1996) [reply package.json]: Ah cool, didn't see that this was in `devDependencies` 👍.

File: CHANGELOG.md (hunk 1 above)
Comment (jsalvata) [reply CHANGELOG.md]: Eli says this repo doesn't have semantic release yet.

File: src/find.js [hunk 4]
```diff
@@ -39,7 +40,9 @@ module.exports = async function(collection, params) {
   // https://www.npmjs.com/package/mongoist#cursor-operations
   const findMethod = collection.findAsCursor ? 'findAsCursor': 'find';
 
-  const results = await collection[findMethod]({ $and: [cursorQuery, params.query] }, params.fields)
```
Comment (edneyreis) [inline src/find.js:42]: Plz, sir. 
could you answer why did you remove the `await`?
I'm getting an error in line 45:
"collatedQuery.sort is not a function"

it happens because `query ` is a Promise

Comment (jsalvata) [reply src/find.js:42]: It wasn't removed. It was moved to line 45.
`query` should be a `FindCursor`: [see docs](https://mongodb.github.io/node-mongodb-native/4.3/classes/Collection.html#find).
It's `toArray` that turns the `FindCursor` into a Promise: [docs](https://mongodb.github.io/node-mongodb-native/4.3/classes/FindCursor.html#toArray).
